### PR TITLE
Add small variant to InputSelect

### PR DESCRIPTION
**Pull request by raymondtanwl**
**Changes**
Add small variant to InputSelect

**Changelog entry**
Add small variant to InputSelect

**Additional information**

- You may refer to this [[BOOKINGSG-5642](https://jira.ship.gov.sg/browse/BOOKINGSG-5642)]

### CODE REVIEW

File: src/shared/dropdown-list/dropdown-list.styles.tsx [hunk 1]
```diff
@@ -246,10 +272,20 @@ export const ResultStateContainer = styled.div`
 `;
 
 export const ResultStateText = styled(Text.Body)``;
+export const ResultStateTextSmall = styled(Text.BodySmall)``;
 
-export const LabelIcon = styled(ExclamationCircleFillIcon)`
+export const LabelIcon = styled(ExclamationCircleFillIcon)<IconProps>`
+    ${(props) => {
+        let _size = 1.5;
```
Comment (qroll) [inline src/shared/dropdown-list/dropdown-list.styles.tsx:279]: no need to prefix variables

```suggestion
        let size = 1.5;
```

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/shared/dropdown-list/dropdown-list.styles.tsx [hunk 2]
```diff
@@ -87,6 +98,19 @@ export const ListItem = styled.li<ListItemProps>`
     }}
 `;
 
+const _getMinHeight = (props: ListItemSelectorProps) => {
```
Comment (qroll) [inline src/shared/dropdown-list/dropdown-list.styles.tsx:101]: do update the other instances, the prefix is not needed

```suggestion
const getMinHeight = (props: ListItemSelectorProps) => {
```

File: src/shared/dropdown-list/dropdown-list.styles.tsx [hunk 3]
```diff
@@ -87,6 +98,19 @@ export const ListItem = styled.li<ListItemProps>`
...
+    let _minHeight = 3.5;
```
Comment (qroll) [inline src/shared/dropdown-list/dropdown-list.styles.tsx:102]:
```suggestion
    let minHeight = 3.5;
```

File: src/shared/dropdown-list/dropdown-search.styles.tsx [hunk 4]
```diff
@@ -12,29 +18,53 @@ export const Container = styled.li`
     align-items: center;
 `;
 
-export const SearchInput = styled(BasicInput)`
-    height: 3rem;
+const _commonSearchInputStyle = `
```
Comment (qroll) [inline src/shared/dropdown-list/dropdown-search.styles.tsx:21]: can revert this change since it's only used in 1 place

File: src/shared/dropdown-wrapper/dropdown-wrapper.styles.tsx [hunk 5]
```diff
@@ -18,49 +18,67 @@ export interface DropdownWrapperStyleProps {
 
 export interface ValueLabelStyleProps {
     truncateType?: TruncateType;
+    variant?: DropdownVariantType | undefined;
+}
+
+export interface SelectorStyleProps {
+    variant?: DropdownVariantType;
```
Comment (qroll) [inline src/shared/dropdown-wrapper/dropdown-wrapper.styles.tsx:25]: for non-HTML attributes, prefix them with `$` to mark them as transient props

please update for the other style interfaces

File: src/shared/dropdown-wrapper/dropdown-wrapper.styles.tsx [hunk 6]
```diff
@@ -18,49 +18,67 @@ export interface DropdownWrapperStyleProps {
...
-    border-radius: ${BORDER_RADIUS};
-    border: none;
-    background: transparent;
-
-    :focus,
-    :active {
-        outline: none;
-    }
+export const baseSelectorCSS = (props?: SelectorStyleProps) => {
+    const height = _getHeight(props.variant);
```
Comment (qroll) [inline src/shared/dropdown-wrapper/dropdown-wrapper.styles.tsx:50]: instead of converting to a function, `css` can also take in props

```
export const baseSelectorCSS = css<SelectorStyleProps>`
  ${props => ...}
`
```

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/shared/dropdown-wrapper/dropdown-wrapper.styles.tsx (hunk 5 above)
Comment (qroll) [reply src/shared/dropdown-wrapper/dropdown-wrapper.styles.tsx]: whoops sorrry if it wasn't clear

for component props, there's no need to prefix them with `$`

however for styled components, `$` is needed to mark them as transient, so that extra non-standard attributes are not passed down to the underlying HTML element

Comment (raymondtanwl) [reply src/shared/dropdown-wrapper/dropdown-wrapper.styles.tsx]: @qroll i have pushed a commit to remove `$` from the component props.
you can have a look again :)

File: src/shared/dropdown-list/dropdown-list.styles.tsx [hunk 7]
```diff
@@ -245,11 +277,23 @@ export const ResultStateContainer = styled.div`
...
+        TextStyleHelper.getTextStyle(
+            props.$variant === "small" ? "BodySmall" : "Body",
+            "regular"
+        )}
+`;
 
-export const LabelIcon = styled(ExclamationCircleFillIcon)`
+export const LabelIcon = styled(ExclamationCircleFillIcon)<IconProps>`
+    ${(props) => {
+        // 1rem 16px, 1.5rem 24px
```
Comment (qroll) [inline src/shared/dropdown-list/dropdown-list.styles.tsx:290]: no need to add the comments on the unit conversion here, as the rem->px is a standard convention in this repo

File: src/shared/dropdown-list/dropdown-list.styles.tsx [hunk 8]
```diff
@@ -245,11 +277,23 @@ export const ResultStateContainer = styled.div`
     align-items: center;
 `;
 
-export const ResultStateText = styled(Text.Body)``;
+export const ResultStateText = styled.text<ResultStateProps>`
```
Comment (qroll) [inline src/shared/dropdown-list/dropdown-list.styles.tsx:280]: I don't think text is a standard html element? render a div instead

```suggestion
export const ResultStateText = styled.div<ResultStateProps>`
```

File: src/shared/dropdown-list/dropdown-list.tsx [hunk 9]
```diff
@@ -460,19 +468,29 @@ export const DropdownList = <T, V>({
...
+
             return (
                 <ResultStateContainer key="loading" data-testid="list-loading">
-                    <Spinner $buttonStyle="secondary" size={24} />
-                    <ResultStateText>Loading...</ResultStateText>
+                    <Spinner $buttonStyle="secondary" size={spinnerSize} />
+                    <ResultStateText $variant={variant}>
+                        Loading...
+                    </ResultStateText>
+                    ;
```
Comment (qroll) [inline src/shared/dropdown-list/dropdown-list.tsx:493]: unintended semicolon

File: src/shared/dropdown-list/dropdown-search.styles.tsx [hunk 10]
```diff
@@ -12,29 +24,43 @@ export const Container = styled.li`
...
     color: ${Color.Neutral[3]};
     cursor: pointer;
+    ${(props) => {
+        if (props.$variant === "small")
+            return css`
+                svg {
+                    ${getCssHeightAndWidth(props.$variant)}
+                }
+            `;
+    }}
```
Comment (qroll) [inline src/shared/dropdown-list/dropdown-search.styles.tsx:58]: actually this is redundant as the CancelIcon has its own css styling already

File: src/shared/dropdown-wrapper/dropdown-wrapper.styles.tsx [hunk 11]
```diff
@@ -18,27 +18,42 @@ export interface DropdownWrapperStyleProps {
...
+
+export interface SelectorStyleProps {
+    $variant?: DropdownVariantType | undefined;
 }
 
 // =============================================================================
 // STYLING
 // =============================================================================
 const BORDER_RADIUS = "4px";
+const getHeight = (variant?: DropdownVariantType | undefined) => {
```
Comment (qroll) [inline src/shared/dropdown-wrapper/dropdown-wrapper.styles.tsx:32]: small nitpick but can you add a newline between 31-32? to separate the two items

**Review by qroll — CHANGES_REQUESTED** [review]
just some minor code style comments

separately, can I trouble you to squash your commits in this PR? there were various reverts done here, so this is to keep the commit history clean and make it easier to compare diffs with older code

File: src/shared/dropdown-list/dropdown-search.styles.tsx (hunk 10 above)
Comment (raymondtanwl) [reply src/shared/dropdown-list/dropdown-search.styles.tsx:56]: i added this because its ClickableIcon set default svg height and width to 1.125rem
https://github.com/LifeSG/react-design-system/blob/55f2a59308303af3298c7c44a1ad0ba05e713eed/src/shared/clickable-icon/clickable-icon.style.tsx#L23

hmm shall i modify ClickableIcon to have the variant prop too?

Comment (qroll) [reply src/shared/dropdown-list/dropdown-search.styles.tsx:56]: then in that case, CancelButton can handle the `svg` styling for both variants, and CancelIcon can be removed entirely (the component can use CrossIcon directly instead)

I'd say ClickableIcon doesn't need a variant for now because there's actually no standardised sizing for icons

**Comment by qroll** [comment]
with the latest changes, the variant prop doesn't seem to be available

**Comment by raymondtanwl** [comment]
> with the latest changes, the variant prop doesn't seem to be available

as discussed, I have removed variant prop from DropdownStyleProps and instead added it specifically on InputSelectProps, InputMultiSelectProps and DropdownListProps

**Review by qroll — APPROVED** [review]
